server/src/test/java/org/elasticsearch/snapshots/SharedClusterSnapshotRestoreIT.java

```diff
@@ -2961,6 +2961,103 @@ public void testRestoreSnapshotWithCorruptedIndexMetadata() throws Exception {
         assertAcked(client().admin().cluster().prepareDeleteSnapshot("test-repo", snapshotInfo.snapshotId().getName()).get());
     }
 
+    /**
+     * Tests that a shard snapshot with a corrupted shard index file can still be used for restore and incremental snapshots.
+     */
+    public void testSnapshotWithCorruptedShardIndexMetadata() throws Exception {
+        final Client client = client();
+        final Path repo = randomRepoPath();
+        final String indexName = "test-idx";
+
+        assertAcked(prepareCreate(indexName).setSettings(Settings.builder()
+            .put(SETTING_NUMBER_OF_SHARDS, 1).put(SETTING_NUMBER_OF_REPLICAS, 0)));
+
+        final int nDocs = randomIntBetween(1, 10);
```

Comment:
In general, I find it useful to log each logical step in the test. This serves two purposes:
* It's easier to follow the code, because log messages serve as comments
* When figuring out why the test has failed, you can use this log messages as anchors.

Log messages could be like:
* Create index and index some documents
* Create repository
* Create snapshot
* Truncate shard index file
* Delete index
* Restore the snapshot and check that documents are there
and so on...